R/modules.R

```diff
@@ -277,25 +291,50 @@ module <- function(label = "module",
   checkmate::assert_function(server)
   checkmate::assert_function(ui)
   checkmate::assert_character(filters, min.len = 1, null.ok = TRUE, any.missing = FALSE)
-  checkmate::assert_list(server_args, null.ok = TRUE)
-  checkmate::assert_list(ui_args, null.ok = TRUE)
+  checkmate::assert_list(server_args, null.ok = TRUE, names = "named")
+  checkmate::assert_list(ui_args, null.ok = TRUE, names = "named")
+
+  server_formals <- names(formals(server))
+  if (!any(c("id", "input", "output", "session") %in% server_formals)) {
```

Comment:
So something like `function(input, x)` would be allowed - I guess that's fine as if you are silly enough to do that then shiny will shout loudly at you.

Reply:
Fine, I'm changing to
```r
!(
    "id" %in% server_formals ||
    all(c("input", "output", "session") %in% server_formals)
  )
```

I will include test also